encoding/rs/decode.go

```diff
@@ -24,16 +24,27 @@ func (e *Encoder) Decode(frames []FrameCoeffs, indices []uint64, maxInputSize ui
 		return nil, err
 	}
 
-	numSys := encoding.GetNumSys(maxInputSize, g.ChunkLength)
+	if len(frames) != len(indices) {
+		return nil, errors.New("number of frames must equal number of indices")
+	}
+
+	// Remove duplicates
+	frameMap := make(map[uint64]FrameCoeffs, len(indices))
+	for i, index := range indices {
```

Comment:
nit: having loop variables named both `i` and `index` is confusing- consider renaming `index` to something like `frameIndex` to help remove ambiguity